File: Iter.Api/Iter.Core/EntityModelss/Reservation.cs
namespace Iter.Core.EntityModels
{
    public class Reservation : BaseEntity
    {
        public string ReservationNumber { get; set; }

        public string DeparturePlace { get; set; }

        public string Reminder { get; set; }
        
        public Guid ClientId { get; set; }
        
        public Client Client { get; set; }

        public Guid ArrangmentId { get; set; }

        public Arrangement Arrangement { get; set; }

        public int ReservationStatusId { get; set; }

        public ReservationStatus ReservationStatus { get; set; }

        public Guid ArrangementPriceId { get; set; }

        public ArrangementPrice ArrangementPrice { get; set; }

        public decimal TotalPaid { get; set; }

        public int Rating { get; set; }
    }
}

File: Iter.Api/ui/Iter.Mobile/lib/models/reservation.dart
import 'package:iter_mobile/models/arrangement_price.dart';
import 'package:iter_mobile/models/arrangment.dart';
import 'package:iter_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final String id;
  final String? arrangementPriceId;
  final String? reservationNumber;
  final String? departurePlace;
  final User? user;
  final Arrangement? arrangement;
  final ArrangementPrice? arrangementPrice;
  final int? reservationStatusId;
  final String? reservationStatusName;
  final double? totalPaid;
  final int? rating;
  final String? reminder;

  Reservation(
      {
      required this.id,
      required this.reservationNumber,
      required this.arrangementPriceId,
      required this.departurePlace,
      required this.user,
      required this.arrangement,
      required this.arrangementPrice,
      required this.reservationStatusId,
      required this.reservationStatusName,
      required this.totalPaid,
      required this.rating,
      required this.reminder
      });

  factory Reservation.fromJson(Map<String, dynamic> json) => _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
